python/ray/serve/_private/deployment_state.py

```diff
@@ -117,6 +118,30 @@ def from_deployment_info(
 
         return cls(info, num_replicas, version, deleting)
 
+    def compare_state_excluding_num_replicas(self, other_target_state) -> bool:
+        """Compare with another target state.
+
+        Returns: Whether all configurable info is identicable other than num_replicas.
+        """
+
+        if not other_target_state.info:
+            return False
+
+        return (
+            self.info.replica_config.ray_actor_options
+            == other_target_state.info.replica_config.ray_actor_options
+            and self.info.replica_config.placement_group_bundles
+            == other_target_state.info.replica_config.placement_group_bundles
+            and self.info.replica_config.placement_group_strategy
+            == other_target_state.info.replica_config.placement_group_strategy
+            and self.info.replica_config.max_replicas_per_node
+            == other_target_state.info.replica_config.max_replicas_per_node
+            and self.info.deployment_config.dict(exclude={"num_replicas"})
+            == other_target_state.info.deployment_config.dict(exclude={"num_replicas"})
+            and self.version
+            and self.version == other_target_state.version
```

Comment:
This is intentional because `version` can theoretically be `None`, and in that case we generate a random version on the spot. In reality I think that code path is only used for v1 deployments because I cleaned up all uses of passing in version=None for v2, but I haven't gotten around to cleaning up the code in deployment state manager yet.

Reply:
Got it, could you add a comment mentioning that?